op-challenger/metrics/metrics.go

```diff
@@ -120,3 +134,17 @@ func (m *Metrics) RecordGameStep() {
 func (m *Metrics) RecordCannonExecutionTime(t float64) {
 	m.cannonExecutionTime.Observe(t)
 }
+
+func (m *Metrics) RecordGameInProgress() {
+	m.trackedGames.WithLabelValues("in_progress").Inc()
+}
+
+func (m *Metrics) RecordGameDefenderWon() {
+	m.trackedGames.WithLabelValues("in_progress").Dec()
+	m.trackedGames.WithLabelValues("defender_won").Inc()
+}
+
+func (m *Metrics) RecordGameChallengerWon() {
+	m.trackedGames.WithLabelValues("in_progress").Dec()
+	m.trackedGames.WithLabelValues("challenger_won").Inc()
+}
```

Comment:
This is going to count games multiple times on every update.  We need a single method that accepts the total counts for each game type and then just set those values on the gauge.

Reply:
Right, good callout - I'm not really seeing where this atomic update would happen. The coordinator just routes jobs individually and never has a complete view of games with updated game types as far as I understand.

Reply:
The coordinator always gets a full list of games - that's how it can produce the list of still-in-progress games that it shouldn't delete files for.  So the coordinator `schedule` method could compute the totals and make the call to update metrics at the end.  Its `createJob` method should probably just be updated to return the job to execute (possibly null as it is now) plus also return the current status as its recorded in the `gameState`. I wouldn't make another call to fetch the game status, just depend on it being updated whenever the game progresses.

Reply:
ah yeah admittedly the delete code didn't actually wind up in the `schedule` method which makes it harder to see that - it does remove game states for any games not in the list which also depends on the fact that its given the full list of active games.

It does have the allowlist already applied so there are some games it doesn't "see" but I think that's probably good - then the metric reports only the games that the challenger is actually tracking.

Reply:
Ok attempted to change it up to this - let me know if this is along the lines of what you're thinking